snippets/elo.md

```diff
@@ -15,17 +15,20 @@ const elo = ([...ratings], kFactor = 32, selfRating) => {
   const expectedScore = (self, opponent) => 1 / (1 + 10 ** ((opponent - self) / 400));
   const newRating = (rating, i) =>
     (selfRating || rating) + kFactor * (i - expectedScore(i ? a : b, i ? b : a));
+    
   if (ratings.length === 2) {
     return [newRating(a, 1), newRating(b, 0)];
-  } else {
-    for (let i = 0; i < ratings.length; i++) {
-      let j = i;
-      while (j < ratings.length - 1) {
-        [ratings[i], ratings[j + 1]] = elo([ratings[i], ratings[j + 1]], kFactor);
-        j++;
-      }
+  } 
+  
+  for (let i = 0, l = ratings.length; i < l; i++) {
+    let j = i;
+    
+    while (j < l - 1) {
+      [ratings[i], ratings[j + 1]] = elo([ratings[i], ratings[j + 1]], kFactor);
+      j++;
```

Comment:
If we're constantly using j+1 we should just `j++` before we execute this code or `++j` on the first call so we don't have to constantly put j+1

Reply:
Yeah, `j++` seems logical.

Reply:
> There's more that can be done in terms of readability, but a modest speed improvement and a proper guard statement is appreciated.

I agree with this one. Speaking about readability this snippet is categorized as advanced as far as I remember so readability was always a bit tough here. We assume that more experienced devs are going to read these anyways so ... Let's leave readability to some other PR like @skatcat31  mentioned.

Thanks for contributing!